packages/btp-utils/src/destination.ts

```diff
@@ -107,3 +116,13 @@ export function isFullUrlDestination(destination: Destination): boolean {
             isGenericODataDestination(destination)
     );
 }
+
+/**
+ * Checks whether the provided destination is configured as an on-premise system.
+ *
+ * @param destination destination info
+ * @returns true if the destination is configured as an on-premise system
+ */
+export function isOnPremiseSystem(destination: Destination): boolean {
```

Comment:
So we are dealing with a spectic type, so make this `isOnPremiseDestination` since destination objects are only related to BTP CF destinations. In Fiori tools, a system meant different things!

Reply:
Also, you need to ensure you handling the scenerio if ProxyType is not returned;
```
export function isOnPremiseDestination(destination: Destination): boolean {
    return Boolean(destination.ProxyType?.includes(ProxyType.ON_PREMISE));
}
```